packages/core/src/rum/instrumentation/resourceTracking/helpers/FormattedUUID.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import type { BigInteger } from 'big-integer';
+
+import { PropagatorType } from '../../../types';
+import type {
+    TraceId,
+    SpanId,
+    TracingIdentifier
+} from '../distributedTracing/TracingIdentifier';
+import {
+    TracingIdType,
+    TracingIdFormat
+} from '../distributedTracing/TracingIdentifier';
+
+export class FormattedUUID {
+    /**
+     * Read-only generated ID as a {@link BigInteger}.
+     */
+    get id(): BigInteger {
+        return this.uuid.id;
+    }
+
+    /**
+     * Read-only type to determine whether the identifier is a {@link TraceId} or a {@link SpanId}.
+     */
+    get type(): TracingIdType {
+        return this.uuid.type;
+    }
+
+    private static readonly DEFAULT_PROPAGATOR_TYPE =
+        PropagatorType.TRACECONTEXT;
+
+    private uuid: TracingIdentifier;
+
+    public constructor(uuid: TraceId | SpanId) {
+        this.uuid = uuid;
+    }
+
+    /**
+     * Returns a string representation of the Tracing ID.
+     * @param format - The type of representation to use.
+     * @returns The ID as a string in the specified representation type.
+     */
+    toString(format: TracingIdFormat): string {
+        return this.uuid.toString(format);
+    }
+
+    /**
+     * Returns the string representation of the UUID that can be used for resources.
+     * @returns the formatted UUID.
+     */
+    public forResource(): string {
```

Comment:
suggestion: maybe it is worth to rename it to `forRumResource`? Otherwise it is quite generic.